tests/cmake_integration/Messenger/Messenger_1/CMakeLists.txt

```diff
@@ -44,7 +44,9 @@ OPENDDS_TARGET_SOURCES(subscriber
 )
 
 foreach(t ${all_targets})
-  target_link_libraries(${t} OpenDDS::OpenDDS)
+  # Use of PRIVATE library scope here is a regression test for
+  # https://github.com/objectcomputing/OpenDDS/issues/1336
+  target_link_libraries(${t} PRIVATE OpenDDS::OpenDDS)
```

Comment:
This is probably due to my lack of in-depth CMake knowledge.  What does it mean for an executable to have a private library dependency?

Also the `foreach` here looks very strange, but I'm used to declarative instead of procedural build definitions.

Reply:
I think that questions is a valid one, because as far as I know it doesn't mean anything. It should just test that CMP0023 is happy.

Reply:
I think that questions is a valid one, because as far as I know it doesn't mean anything. It should just test that CMP0023 is happy.

Reply:
Let's move that elsewhere, since this file is essentially here as documentation of how the CMake support should be used.

Reply:
I'm not sure where else this would go. Also I'm not so sure this is a good CMake example given we're copying the messenger test files here. That's part of the reason I added a CMakeLists.txt to the devguide messenger in #1287.

Reply:
Could we just remove it then?

Reply:
Made it PUBLIC scope as discussed.